### PR TITLE
Add utility fns to convert from magma to hwtypes

### PR DESCRIPTION
**Pull request by leonardt**
Intended usecase is for fault where we inspect the magma circuit type
and automatically martial values into the corresponding hwtypes type for
simulation.

Note for now we don't support nested arrays since we don't have an
analagous hwtypes type.  In theory we could use numpy arrays and set the
datatype to be the leaf hwtypes type.

### CODE REVIEW

File: magma/hwtypes_convert.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    elif issubclass(T, m.Bits):
+        return ht.BitVector[T.N]
+    elif issubclass(T, m.Product):
+        _fields = {k: to_hwtypes(v) for k, v in T.field_dict.items()}
+        return ht.Product.from_fields(T.__name__, _fields)
+    elif issubclass(T, m.Tuple):
+        _fields = tuple(to_hwtypes(v) for v in T.fields)
+        return ht.Tuple[_fields]
+    else:
+        raise NotImplementedError(T)
```
Comment (rsetaluri) [inline magma/hwtypes_convert.py:23]:
```suggestion
import hwtypes as ht

from magma.t import Kind
from magma.bit import Bit
from magma.bits import Bits, SInt, UInt
from magma.tuple import Product, Tuple


def to_hwtypes(T: m.Kind):
    if not isinstance(T, m.Kind):
        raise TypeError("Expected a magma type")
    if issubclass(T, m.Bit):
        return ht.Bit
    if issubclass(T, m.SInt):
        return ht.SIntVector[T.N]
    if issubclass(T, m.UInt):
        return ht.UIntVector[T.N]
    if issubclass(T, m.Bits):
        return ht.BitVector[T.N]
    if issubclass(T, m.Product):
        _fields = {k: to_hwtypes(v) for k, v in T.field_dict.items()}
        return ht.Product.from_fields(T.__name__, _fields)
    if issubclass(T, m.Tuple):
        _fields = tuple(to_hwtypes(v) for v in T.fields)
        return ht.Tuple[_fields]
    raise NotImplementedError(T)
```

Comment (rsetaluri) [reply magma/hwtypes_convert.py:23]: I also think this function can go into `magma/type_utils.py`

File: tests/test_hwtypes_convert.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import magma as m
+import hwtypes as ht
+from magma.hwtypes_convert import to_hwtypes
```
Comment (rsetaluri) [inline tests/test_hwtypes_convert.py:3]:
```suggestion
from magma.type_utils import to_hwtypes
```

**Review by rsetaluri — CHANGES_REQUESTED** [review]

File: magma/type_utils.py [hunk 3]
```diff
@@ -2,11 +2,15 @@
 import dataclasses
 import typing
 
+import hwtypes as ht
+
+from magma.bit import Bit
 from magma.array import Array
```
Comment (rsetaluri) [inline magma/type_utils.py:8]:
```suggestion
from magma.array import Array
from magma.bit import Bit
```

**Review by rsetaluri — APPROVED** [review]
last minor formatting

**Review by cdonovick — APPROVED** [review]

**Pull request merged**